dj/sql/functions.py

```diff
@@ -1277,6 +1277,89 @@ def infer_type(  # noqa: F811  # pragma: no cover
     return ct.BooleanType()  # pragma: no cover
 
 
+class ArrayDistinct(Function):  # pylint: disable=abstract-method
+    """
+    array_distinct(array) - Removes duplicate values from the array.
+    """
+
+
+@ArrayDistinct.register
+def infer_type(  # noqa: F811
+    array: ct.ListType,
+) -> ct.ListType:
+    return array.type
+
+
+class ArrayExcept(Function):  # pylint: disable=abstract-method
+    """
+    array_except(array1, array2) - Returns an array of the elements in
+    array1 but not in array2, without duplicates.
+    """
+
+
+@ArrayExcept.register
+def infer_type(  # noqa: F811
+    array1: ct.ListType,
+    array2: ct.ListType,
+) -> ct.ListType:
+    return array1.type
+
+
+class ArrayIntersect(Function):  # pylint: disable=abstract-method
```

Comment:
Doesn't have to be done in this PR but I'm realizing we should probably put this `# pylint: disable=abstract-method` once at the top of the file.

Reply:
Good point! I was going to try to arrange these functions in alphabetical order as well, so I'll do that together in a separate PR

Reply:
@samredai did that in https://github.com/DataJunction/dj/pull/560